validator/src/bootstrap.rs

```diff
@@ -1219,47 +1265,30 @@ mod with_incremental_snapshots {
             if bootstrap_config.no_untrusted_rpc
                 && !is_trusted_validator(&rpc_peer.id, &validator_config.trusted_validators)
             {
-                // Ignore all untrusted peers
+                // We were told to ignore untrusted peers
                 continue;
             }
 
-            cluster_info
-                .get_incremental_snapshot_hashes_for_node(&rpc_peer.id)
-                .and_then(
-                    |crds_value::IncrementalSnapshotHashes { base, hashes, .. }| {
-                        // Newer hashes are pushed to the end of `hashes`, so the last element should
-                        // be the newest (i.e. have the highest slot).
-                        //
-                        // NOTE: It is important that the result of `last().map()` is the return  value
-                        // from `and_then()`, so that the `or_else()` can run in _both_ scenarios where
-                        // either (1) the peer does not have incremental snapshots enabled, or (2) the
-                        // peer has not generated any incremental snapshots yet.
-                        hashes.last().map(|incremental_snapshot_hash| {
-                            peer_snapshot_hashes.push(PeerSnapshotHash {
-                                rpc_contact_info: rpc_peer.clone(),
-                                snapshot_hash: SnapshotHash {
-                                    full: base,
-                                    incr: Some(*incremental_snapshot_hash),
-                                },
-                            })
-                        })
-                    },
-                )
-                // If the peer does not have any incremental snapshot hashes, then get its highest full
-                // snapshot hash instead.
-                .or_else(|| {
-                    cluster_info.get_snapshot_hash_for_node(&rpc_peer.id, |hashes| {
-                        if let Some(full_snapshot_hash) = hashes.last() {
-                            peer_snapshot_hashes.push(PeerSnapshotHash {
-                                rpc_contact_info: rpc_peer.clone(),
-                                snapshot_hash: SnapshotHash {
-                                    full: *full_snapshot_hash,
-                                    incr: None,
-                                },
-                            })
-                        }
-                    })
+            // Get the highest incremental snapshot hash for this peer.
```

Comment:
Nit: this comment is also too chatty.  "// Get the highest incremental snapshot hash for this peer." over a line of code that calls a function named `get_highest_incremental_snapshot_hash_for_peer()`

Reply:
Will remove.